src/lightning/fabric/strategies/deepspeed.py

```diff
@@ -32,12 +32,22 @@
 from lightning.fabric.strategies.ddp import DDPStrategy
 from lightning.fabric.strategies.strategy import _Sharded
 from lightning.fabric.utilities.distributed import log
+from lightning.fabric.utilities.imports import _TORCH_GREATER_EQUAL_2_0
 from lightning.fabric.utilities.rank_zero import rank_zero_info, rank_zero_only, rank_zero_warn
 from lightning.fabric.utilities.seed import reset_seed
 from lightning.fabric.utilities.types import _PATH
 
-# check packaging because of https://github.com/microsoft/DeepSpeed/pull/2771
-_DEEPSPEED_AVAILABLE = RequirementCache("deepspeed") and RequirementCache("packaging>=20.0")
+_DEEPSPEED_AVAILABLE = (
+    # DeepSpeed fails under 0.8.2 with torch 2.0: https://github.com/microsoft/DeepSpeed/pull/2863
+    RequirementCache("deepspeed>=0.8.2")
+    or not _TORCH_GREATER_EQUAL_2_0
+    and (
+        RequirementCache("deepspeed")
+        # check packaging because of https://github.com/microsoft/DeepSpeed/pull/2771
+        # remove the packaging check when min version is >=0.8.1
+        and RequirementCache("packaging>=20.0")
+    )
```

Comment:
```suggestion
    # check packaging because of https://github.com/microsoft/DeepSpeed/pull/2771
    # remove the packaging check when min version is >=0.8.1
    and RequirementCache("deepspeed") and RequirementCache("packaging>=20.0")
```